### DISCUSSION TITLE
How to document types

### DISCUSSION THREAD

**Discussion by Timmmm** [cue-lang/cue#2215] [General]
One of the really nice things about XML schema is that you can add documentation for types. This is supported by IDEs and allows you to hover on elements to see help text about them. It's *extremely* useful. I think JSON Schema provides [a similar feature](https://json-schema.org/draft/2020-12/json-schema-validation.html#name-title-and-description) (but I haven't used it and don't know how good IDE support it).

As far as I can tell there's no way to do that in Cue?

**Comment by verdverm** [comment]
There are plans for a language server to enable LSP support in IDEs. The docs are already attached to values in the AST, where appropriate. They work pretty similar to Go type documentation. I suspect there will be a CLI command too, like Go

**Comment by Timmmm** [comment]
How do you actually write documentation in a `.cue` file though? Or is it going to use heuristics to turn the `// comments` into documentation?

**Reply by slewiskelly** [reply]
Yes, you'd just write comments.

You can read more about Godoc [here](https://go.dev/blog/godoc). The planned LSP, I believe, is going to be based on Dagger, which is [here](https://github.com/dagger/cuelsp), and can see in action [here](https://github.com/dagger/cuelsp/blob/main/docs/vim.md#44-use-vim-lsp-with-dagger).
